### PR TITLE
feat(tdhttp): allow usage of td.Q when using PostForm helpers

### PR DESCRIPTION
**Pull request by c-roussel**

### CODE REVIEW

File: helpers/tdhttp/request.go [hunk 1]
```diff
@@ -336,6 +341,15 @@ func Post(target string, body io.Reader, headersQueryParams ...any) *http.Reques
 	return req
 }
 
+// URLValuesEncoder is an interface PostForm data must implement.
+// Encode can be called to generate a "URL encoded" form such as
+// ("bar=baz&foo=quux") sorted by key.
+//
+// url.Values and tdhttp.Q implement this interface.
```
Comment (maxatome) [inline helpers/tdhttp/request.go:348]:
```suggestion
// [url.Values] and [Q] implement this interface.
```

File: helpers/tdhttp/request.go [hunk 2]
```diff
@@ -336,6 +341,15 @@ func Post(target string, body io.Reader, headersQueryParams ...any) *http.Reques
 	return req
 }
 
+// URLValuesEncoder is an interface PostForm data must implement.
```
Comment (maxatome) [inline helpers/tdhttp/request.go:344]:
```suggestion
// URLValuesEncoder is an interface [PostForm] and [TestAPI.PostForm] data
// must implement. …
```

**Review by maxatome — CHANGES_REQUESTED** [review]

File: helpers/tdhttp/request_test.go [hunk 3]
```diff
@@ -256,6 +256,30 @@ func TestNewRequest(tt *testing.T) {
...
+				},
+			},
+			td.StructFields{
+				"URL": td.String("/path"),
+				"Body": td.Smuggle(
+					io.ReadAll,
+					[]byte("param1=val1&param2=val2"),
+				),
+			}))
+
```
Comment (maxatome) [inline helpers/tdhttp/request_test.go:282]: Could you add the `nil` case?
```go
tdhttp.PostForm("/path", nil)
```

Comment (c-roussel) [reply helpers/tdhttp/request_test.go:282]: test added

**Comment by maxatome** [comment]
Thanks for your contribution @c-roussel and welcome as new go-testdeep contributor! :)

**Pull request merged**